www/js/control.js

```diff
@@ -0,0 +1,431 @@
+        }
+    }
+
+    $scope.userDataSaved = function() {
+        return UserCalorieData.get().userDataSaved == true;
+    }
+    if ($scope.userDataSaved()) {
+        $scope.getUserData();
+    }
+    $scope.getLowAccuracy = function() {
+        //  return true: toggle on; return false: toggle off.
+        if ($scope.settings.collect.config == null) {
+            return false; // config not loaded when loading ui, set default as false
+        } else {
+            var accuracy = $scope.settings.collect.config.accuracy;
+            var v;
+            for (var k in $scope.settings.collect.accuracyOptions) {
+                if ($scope.settings.collect.accuracyOptions[k] == accuracy) {
+                    v = k;
+                    break;
+                }
+            }
+            if ($scope.isIOS()) {
+                return v != "kCLLocationAccuracyBestForNavigation" && v != "kCLLocationAccuracyBest" && v != "kCLLocationAccuracyTenMeters";
+            } else if ($scope.isAndroid()) {
+                return v != "PRIORITY_HIGH_ACCURACY";
+            }
+
+        }
+    }
+    $scope.toggleLowAccuracy = function() {
+        $scope.settings.collect.new_config = JSON.parse(JSON.stringify($scope.settings.collect.config));
+        if ($scope.getLowAccuracy()) {
+            if ($scope.isIOS()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["kCLLocationAccuracyBest"];
+            } else if ($scope.isAndroid()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["PRIORITY_HIGH_ACCURACY"];
+            }
+        } else {
+            if ($scope.isIOS()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["kCLLocationAccuracyHundredMeters"];
+            } else if ($scope.isAndroid()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["PRIORITY_BALANCED_POWER_ACCURACY"];
+            }
+        }
+        ControlHelper.dataCollectionSetConfig($scope.settings.collect.new_config)
+        .then(function(){
+            console.log("setConfig Sucess");
+        }, function(err){
+            console.log("setCongif Error: " + err);
+        });
+    }
+    $scope.ionViewBackgroundClass = function() {
+        return ($scope.dark_theme)? "ion-view-background-dark" : "ion-view-background";
+    }
+    $scope.getDarkTheme = function() {
+        return $scope.dark_theme;
+    }
+    $scope.toggleDarkTheme = function() {
+        if ($scope.dark_theme) {
+            $rootScope.dark_theme = false;
+            $scope.dark_theme = false;
+            StartPrefs.setDefaultTheme(null);
+        } else {
+            $rootScope.dark_theme = true;
+            $scope.dark_theme = true;
+            StartPrefs.setDefaultTheme('dark_theme');
+        }
+        $ionicPopup.alert({template: 'Restart the app to see all changes!'})
```

Comment:
You should be able to reload the app automatically (recall that all of this is HTML after all. According to 
http://stackoverflow.com/questions/21885518/angularjs-reload-page

`$window.location.reload();` should work